pkg/bootstrap/platform/gcp.go

```diff
@@ -290,10 +290,11 @@ func zoneToRegion(z string) (string, error) {
 // Locality returns the GCP-specific region and zone.
 func (e *gcpEnv) Locality() *core.Locality {
 	var l core.Locality
+	md := e.metadata
 	if e.shouldFillMetadata() {
-		z, zerr := metadata.Zone()
-		if zerr != nil {
-			log.Warnf("Error fetching GCP zone: %v", zerr)
+		z := md[GCPLocation]
```

Comment:
This seems to rely on `Metadata()` having already been called which may not be true

Reply:
`cloud.google.com/go/compute/metadata` methods are being called only if anything is not present in our env variable which is fine IMO.

Reply:
`md` is empty when Locality() is called sincei t relies on it being populated by the call to `Metadata()`. I don't mean `cloud.google.com/go/compute/metadata`